staging/src/k8s.io/apiserver/pkg/server/options/tracing.go

```diff
@@ -101,11 +101,12 @@ func (o *TracingOptions) ApplyTo(es *egressselector.EgressSelector, c *server.Co
 		if err != nil {
 			return err
 		}
-
-		otelDialer := func(ctx context.Context, addr string) (net.Conn, error) {
-			return egressDialer(ctx, "tcp", addr)
+		if egressDialer != nil {
```

Comment:
what is going to happen if egressDialer is nil?

Reply:
It won't use a custom dialer in that case (it will just use the default dial function).

Reply:
huh... probably worth noting loudly in the Lookup documentation that this can return `nil, nil` (and maybe sweeping for other callers that assume a nil error means a non-nil dialer)